src/main/java/sopt/makers/authentication/application/service/auth/AuthenticateSocialAccountService.java

```diff
@@ -40,9 +41,22 @@ public AuthenticateSocialTokenInfo authenticate(AuthenticateSocialAccountCommand
             user.getId(), null, List.of(new SimpleGrantedAuthority(role.name())));
     String accessToken = jwtAuthAccessTokenProvider.generateJwt(customAuthentication);
     String refreshToken = jwtAuthRefreshTokenProvider.generateJwt(accessToken);
+    boolean isFirstLoginUser = user.getProfile().isFirstLogin();
 
-    return AuthenticateSocialTokenInfo.of(
-        accessToken, refreshToken, user.getProfile().hasProfile());
+    if (isFirstLoginUser) {
+      Profile profile = user.getProfile();
+      Profile updatedProfile =
+          profile.updateProfile(
+              profile.email().orElse(null),
+              profile.phone(),
+              profile.birthday(),
+              profile.profileImage().orElse(null),
+              false);
+      userRepository.update(user, updatedProfile);
+      return AuthenticateSocialTokenInfo.of(accessToken, refreshToken, true);
+    }
+
+    return AuthenticateSocialTokenInfo.of(accessToken, refreshToken, false);
```

Comment:
```suggestion
    if (isFirstLoginUser) {
      Profile profile = user.getProfile();
      Profile updatedProfile =
          profile.updateProfile(
              profile.email().orElse(null),
              profile.phone(),
              profile.birthday(),
              profile.profileImage().orElse(null),
              false);
      userRepository.update(user, updatedProfile);
    }
    return AuthenticateSocialTokenInfo.of(accessToken, refreshToken, isFirstLoginUser);
```

요기 return 문 하나만 반환하게 하면 깔끔해질 것 같아요!